app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,23 @@
+class SessionsController < ApplicationController
+  before_action :redirect_users_to_home, only: [:new, :create]
+
+  def create
+    user = User.find_by_email(params[:email])
```

Comment:
Check https://github.com/rails/rails/blob/4-2-stable/activerecord/lib/active_record/dynamic_matchers.rb, or http://railscasts.com/episodes/400-what-s-new-in-rails-4?view=asciicast (especially @ 6:23 to 6:26 - the implementation isn't very clean for this, meaning it is defining the method on the fly through method missing.) or just another article http://chrisholtz.com/blog/in-defense-of-dynamic-finders/ . They are not deprecated yet, but i wouldn't be surprised if they did at some point. Also I remember Chris mentioning it at 1 live session and saying going through method missing is not that efficient. But for the latter I don't know how to reference it since its only in my memory and i would need to rewatch a lot of live sessions to get that info.